OPHD/States/MapViewStateUi.cpp

```diff
@@ -334,17 +332,8 @@ void MapViewState::populateStructureMenu()
 	}
 	else
 	{
-		mStructures.addItem(constants::Laboratory, 58, StructureID::SID_LABORATORY);
-		mStructures.addItem(constants::Park, 75, StructureID::SID_PARK);
-		mStructures.addItem(constants::UndergroundPolice, 61, StructureID::SID_UNDERGROUND_POLICE);
-		mStructures.addItem(constants::RecreationCenter, 73, StructureID::SID_RECREATION_CENTER);
-		mStructures.addItem(constants::Residence, 55, StructureID::SID_RESIDENCE);
-		mStructures.addItem(constants::UndergroundFactory, 69, StructureID::SID_UNDERGROUND_FACTORY);
-		mStructures.addItem(constants::MedicalCenter, 62, StructureID::SID_MEDICAL_CENTER);
-		mStructures.addItem(constants::Nursery, 77, StructureID::SID_NURSERY);
-		mStructures.addItem(constants::Commercial, 66, StructureID::SID_COMMERCIAL);
-		mStructures.addItem(constants::RedLightDistrict, 76, StructureID::SID_RED_LIGHT_DISTRICT);
-		mStructures.addItem(constants::University, 63, StructureID::SID_UNIVERSITY);
+		fillList(mStructures, mStructureTracker.defaultUndergroundStructures());
+		fillList(mStructures, mStructureTracker.unlockedUndergroundStructures());
```

Comment:
I would assume anything in the "default" list is also "unlocked". Maybe it makes sense to initialize the unlocked list to the default list, and then add to the unlocked list from there. Then you'd only need to add elements from a single "unlocked" list here.

Reply:
I wanted to keep them separate so that a list of structures that have been unlocked due to research can be maintained.

Then again, that can also be done using the `ResearchTracker` and the `StructureItemFromString` table (expanded later) to do the same thing when needed. I have a near compulsive need to cache things like this.

I think I'll go with that. Same as the last comment, will make that a separate change set.